dashboard/src/dashboard.py

```diff
@@ -97,22 +105,89 @@ def run_docstyle_check(repository):
     os.system(command)
 
 
+def percentage(passed, failed):
+    total = passed + failed
+    if total == 0:
+        return "0"
+    perc = 100.0 * failed / total
+    return "{:.0f}".format(perc)
+
+
+def parse_linter_results(filename):
+    source = None
+
+    files = {}
+    passed = 0
+    failed = 0
+
+    with open(filename) as fin:
+        for line in fin:
+            if line.endswith(".py\n"):
+                source = line.strip()
+            if line.endswith("    Pass\n"):
```

Comment:
What about adding `line = line.strip()` as a first line of this `for` loop and then you don't have to check whether the line endswith `\n`.